Tombolo/server/controllers/costMonitoringController.js

```diff
@@ -0,0 +1,245 @@
+const { costMonitoring: CostMonitoring } = require('../models');
+const logger = require('../config/logger');
+const { Op } = require('sequelize');
+const { validationResult } = require('express-validator');
+
+async function createCostMonitoring(req, res) {
+  try {
+    const result = await CostMonitoring.create(req.body);
+    return res.status(201).json({
+      success: true,
+      data: result,
+    });
+  } catch (err) {
+    logger.error('Failed to create cost monitoring', err);
+    return res.status(500).json({ success: false, message: err.message });
+  }
+}
+
+async function updateCostMonitoring(req, res) {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).send('Failed to update cost monitoring');
+  }
+  try {
+    const result = await CostMonitoring.update(req.body, {
+      where: { id: req.body.id },
+    });
+
+    if (result[0] === 0) {
+      return res.status(404).json({
+        success: false,
+        message: 'Cost monitoring not found',
+      });
+    }
+
+    return res.status(200).json({
+      success: true,
+      data: result,
+    });
+  } catch (err) {
+    logger.error('Failed to update cost monitoring', err);
+    return res.status(500).json({ success: false, message: err.message });
+  }
+}
+
+async function getCostMonitorings(req, res) {
```

Comment:
We should scope this query by application ID so users only see monitoring data relevant to the application they’re currently in